ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +342,313 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+      var addTerm = function() {
+        addTermCalled += 1;
+      };
+      var addTermsComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <AddTermsComponent
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            addTerm={addTerm}
+            ref={afterMount}
+          />
+        );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        ''
+      );
+      //test error message
+      addTermsComponentRendered.reportError(
+        "Error occurred while adding new term."
+      );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        'Error occurred while adding new term.'
+      );
+    }
+  );
+
+  QUnit.test('Assert that AddVocabulary work properly',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var saveVocabularyResponse;
+      var done = assert.async();
+      var updateParent = function(data) {
+        saveVocabularyResponse = data;
+      };
+      var afterMount = function(component) {
+        assert.equal(
+          component.state.name,
+          ''
+        );
+        assert.equal(
+          component.state.description,
+          ''
+        );
+        assert.equal(
+          component.state.vocabularyType,
+          'm'
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        var formNode = assertAddVocabularyCommon(assert, component);
+        waitForAjax(1, function() {
+          assert.equal(
+            saveVocabularyResponse.id,
+            vocabulary.id
+          );
+          var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+          inputNodes[0].value = "TestA";
+          inputNodes[1].value = "TestA";
+          inputNodes[2].checked = false;
+          inputNodes[3].checked = true;
+          React.addons.TestUtils.Simulate.submit(formNode);
+          waitForAjax(1, function() {
+            assert.equal(
+              saveVocabularyResponse.id,
+              vocabulary.id
+            );
+            done();
+          });
+        });
+      };
+      React.addons.TestUtils.
+        renderIntoDocument(
+          <AddVocabulary
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            updateParent={updateParent}
+            ref={afterMount}
+          />
+        );
+    }
+  );
+
+  QUnit.test('Assert that ajax fail in AddVocabulary',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var done = assert.async();
+      var saveVocabularyResponse;
```

Comment:
It looks like we're not using `saveVocabularyResponse` in this function so we can remove this variable.